alpha/apps/kaltura/lib/webservices/kSessionUtils.class.php

```diff
@@ -580,7 +580,52 @@ public function getDisableEntitlementForEntry()
 		
 		return $entries;
 	}
-	
+
+	public function getDisableEntitlementForPlaylistEntries()
+	{
+		$entries = array();
+		// foreach privileges group
+		foreach( $this->parsedPrivileges as $privilegeType => $privileges)
+		{
+			if ($privilegeType == self::PRIVILEGE_DISABLE_ENTITLEMENT_FOR_PLAYLIST)
+			{
+				foreach($privileges as $privilege)
+				{
+					$entry = entryPeer::retrieveByPKNoFilter($privilege, null, false);
+					$entries[] = $privilege;
+					if($entry->getType() == entryType::PLAYLIST && self::isValidForPlaylistDisableEntitlement($entry->getMediaType()))
+					{
+						$entry_id_list_str = $entry->getDataContent();
+						$result = myPlaylistUtils::getEntryIdsFromStaticPlaylistString($entry_id_list_str);
+						if($result)
+						{
+							$entries = array_merge($entries, $result);
+						}
+					}
+				}
+			}
+		}
+
+		return $entries;
+	}
+
+	/**
+	 * @param $mediaType
+	 * @return bool
+	 */
+	public static function isValidForPlaylistDisableEntitlement($mediaType)
+	{
+		$result = false;
```

Comment:
Since there is only one type valid , maybe consider shortening this function to a single liner

Reply:
We going to have at least one another valid type once we add the new path playlist type